### ISSUE TITLE
[Proposal] Consider providing an ODF version

### ISSUE THREAD

**Issue by scolytus** [OWASP/ASVS#787]
While Microsoft Word has without question an enormous user base, it might be a good thing for an open community to provide its documents in an open format.

If technically feasibly, it would be nice to also see an ODF version.

But this would really just be a nice to have.

**Comment by elarlang** [comment]
If you want to use document like "it was ment to look like", it makes sense to use PDF. Requirements are available in CSV as plain-text.

I'm just curious - what is an use-case for ODF?

**Comment by scolytus** [comment]
The use-case for ODF is the same as for `.docx`.

**Comment by elarlang** [comment]
> The use-case for ODF is the same as for `.docx`.

... and it is?

**Comment by scolytus** [comment]
I don't know. But I suppose there is one, otherwise there would be no `.docx`.

**Comment by jmanico** [comment]
I don’t think ODF is necessary, but perhaps we could release our content in JSON or build conversion engines to JSON so someone motivated could use that as a base for other doc formats...

--
Jim Manico
@Manicode
Secure Coding Education
+1 (808) 652-3805

**Comment by tghosth** [comment]
Hi @scolytus, in https://github.com/OWASP/ASVS/issues/700 they are discussing a python script to perform all the export. Would that be able to help with the ODF output?

**Comment by scolytus** [comment]
@tghosth I would totally say it only makes sense if it's doable with the automated build chain. If you need to do some manual work for every version then I'd say the benefit is not big enough.

**Comment by tghosth** [comment]
So maybe loop into that issue and see if it can be done.

**Comment by scolytus** [comment]
I'll watch the other issue. I've read through it, looks like some basic work needs to be done first. I'll try to continue once this was done.

**Comment by lfservin** [comment]
I don't think it requieres manual intervention. It would be just one more output from pandoc. It can be trivially implemented as part of the make all target

**Comment by vanderaj** [comment]
pandoc allows the creation of ODF I believe. As long as we have an automated build script that actually works, it would be possible to create it from current markdown. Pagination for PDF is done manually.

**Comment by lfservin** [comment]
The makefile is coming along well. Right now I can build a pdf and a docx without trouble. I'm working on the layout a bit. Odf is just one more target in the makefile.
You can play with the makefile if you want, it's in my fork, in the owasp-makefile branch (lfservin/ASVS)

**Comment by lfservin** [comment]
ODF Target will be tested tomorrow. I hope to close this ticket over the weekend. No magic scripts are needed, just a makefile with the appropriate targets

**Comment by muralito** [comment]
Hi Luis. Your makefile works ok for me. I just need to add the *.yaml and some images to my repo. Thanks.
I also installed a few tetex classes required to the build.

**Comment by lfservin** [comment]
well, the yaml file is the former header.md file. Pandoc needs to be able to better understand what the difference is as to generate nicer documents.
